pkg/pdapi/pd_control.go

```diff
@@ -116,6 +151,17 @@ func PdClientURL(namespace Namespace, clusterName string, scheme string) string
 	return fmt.Sprintf("%s://%s-pd.%s:2379", scheme, clusterName, string(namespace))
 }
 
+// ClusterRefPDClientUrl builds the url of cluster pd client
+func ClusterRefPDClientUrl(namespace Namespace, clusterName string, scheme string, clusterDomain string) string {
+	if len(namespace) == 0 {
+		return fmt.Sprintf("%s://%s-pd:2379", scheme, clusterName)
+	}
+	if len(clusterDomain) == 0 {
+		return fmt.Sprintf("%s://%s-pd.%s:2379", scheme, clusterName, string(namespace))
+	}
+	return fmt.Sprintf("%s://%s-pd-peer.%s.svc.%s:2379", scheme, clusterName, string(namespace), clusterDomain)
```

Comment:
Why do we return `pd-peer` here?

Reply:
@lichunzhu Because in the across Kubernetes case, the service IP is not accessible, but the Pod IP is accessible, so in this case (spec.cluster.clusterDomain is set), we have to return the headless service.